pkg/providers/vsphere/vsphere.go

```diff
@@ -1148,15 +1158,72 @@ func (p *vsphereProvider) InstallCustomProviderComponents(ctx context.Context, k
 	return nil
 }
 
-func (p *vsphereProvider) PostBootstrapDeleteForUpgrade(ctx context.Context) error {
+// PostBootstrapDeleteForUpgrade runs any provider-specific operations after bootstrap cluster has been deleted.
+func (p *vsphereProvider) PostBootstrapDeleteForUpgrade(ctx context.Context, cluster *types.Cluster) error {
+	// Delete the daemonset that was used to disable udp offloading in ubuntu/redhat nodes.
+	logger.V(3).Info("Deleting vsphere-disable-udp-offload daemonset")
```

Comment:
We want to delete after the upgrade finishes? Should this be once we know that it was installed after we install Cilium?

Reply:
We need to wait until after the new nodes generated with the updated template roll out since those will definitely have udp offloading disabled.

So we could technically delete the DS a little earlier in the flow (immediately after the new nodes are out), but that would have required a new method in the Provider interface so I wanted to leverage this existing provider-specific method for cleanup.

Reply:
I see got it, I thought it was just once and then be good to go for the rollout.